src/coreComponents/constitutive/fluid/CO2BrineFluid.hpp

```diff
@@ -37,27 +43,28 @@ namespace geosx
 namespace constitutive
 {
 
-template< typename P1DENS, typename P1VISC, typename P2DENS, typename P2VISC, typename FLASH >
-class MultiPhaseMultiComponentFluid : public MultiFluidBase
+template< typename P1DENS, typename P1VISC, typename P1ENTH, typename P1INTENERGY,
+          typename P2DENS, typename P2VISC, typename P2ENTH, typename P2INTENERGY,
+          typename FLASH >
+class CO2BrineFluid : public MultiFluidBase
```

Comment:
> I'm beginning to think we should once again try simple structs to reduce the boilerplate [...] You probably weren't planning on embarking on another refactoring journey in this PR.

Yesterday it was 36 parameters functions or six dimensional arrays.
Today it's a 9 template parameters class.
Refactoring is faaaar too complicated (I've experienced it).
Things need to be done (like https://github.com/GEOSX/GEOSX/pull/1709 or https://github.com/GEOSX/GEOSX/pull/1750), and it shall take a significant amount of our time.
Thank God we have a solid testing policy! 👍

Reply:
Suggestion highly appreciated @klevzoff! I implemented it, and from what I see so far on Lassen, it fixes the gcc compilation error (formal parameter overflow) that was the pebble in my shoe for weeks in the thermal PR (see for instance [here](https://app.travis-ci.com/github/GEOSX/GEOSX/jobs/557811822)). I may be wrong, so I will push soon to confirm this with the Travis CI tests. Thanks!

Reply:
I implemented what you suggested in a separate file called `PhaseModel.hpp` and reduced the number of templates in the CO2-brine fluid class. It is much cleaner now, thanks for your suggestion.

However, I rejoiced too quickly regarding my longstanding compilation error, it is still present in the fluid update kernel (but just does not appear on Lassen with gcc or clang). The error appears when the update kernel is called from `PVTDriver` as shown in the logs, but also when the update kernel is called from the solver:
```
***/GEOSX_TPL-a634fc6/raja/include/RAJA/policy/cuda/forall.hpp(137): Error: Formal parameter space overflowed (4408 bytes required, max 4096 bytes allowed) in function _ZN4RAJA6policy4cuda4impl18forall_cuda_kernelILm256ENS_9Iterators16numeric_iteratorIllPlEEZN5geosx34CompositionalMultiphaseBaseKernels17FluidUpdateKernel6launchINS1_9cuda_execILm256ELb0EEENS8_12constitutive13CO2BrineFluidINSE_10PhaseModelINSE_8PVTProps19EzrokhiBrineDensityENSH_21EzrokhiBrineViscosityENSH_15NoOpPVTFunctionESK_EENSG_INSH_20SpanWagnerCO2DensityENSH_20FenghourCO2ViscosityESK_SK_EENSH_13CO2SolubilityEE13KernelWrapperEEEvlRKT0_RKN7LvArray9ArrayViewIKdLi1ELi0ElNSV_10ChaiBufferEEES11_S11_RKNSW_ISX_Li2ELi0ElSY_EEEUllE_lEEvT1_SS_T2_
```
To overcome this issue, the only technique I am aware of would be to go back to the size-1 views:
```
arrayView1d< typename PHASE1::KernelWrapper > m_phase1;
arrayView1d< typename PHASE2::KernelWrapper > m_phase2;
```
instead of what I implemented here:
https://github.com/GEOSX/GEOSX/blob/ccb0a2e1aa2a95fc4eef5b22fbe7f287eb00cc6c/src/coreComponents/constitutive/fluid/CO2BrineFluid.hpp#L135-L139
but this is really ugly. Is there a better way to do this? In this [discussion](https://github.com/GEOSX/GEOSX/pull/1325#discussion_r577830229), using a `unique_ptr` was mentioned, would it be cleaner?

Reply:
@francoishamon I actually wrote a response to you weeks ago on the energy balance PR, but I accidentally added it as a review comment, so it was left unpublished :( I've posted it now https://github.com/GEOSX/GEOSX/pull/1225#discussion_r781755352, it's not a definitive answer but some ideas about reducing the size of arguments.

Reply:
@klevzoff Thanks, it worked, this PR compiles now, this is great.